File: OlderCodes/JavaCodes/Complex.java
/* From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
   by RH Landau, MJ Paez, and CC BORDEIANU 
   Copyright Princeton University Press, Princeton, 2008.
   Electronic Materials copyright: R Landau, Oregon State Univ, 2008;
   MJ Paez, Univ Antioquia, 2008; and CC BORDEIANU, Univ Bucharest, 2008.
   Support by National Science Foundation                              
   */
// Complex.java:   Creates "Complex" class with static members

public class Complex {
  public double re, im;                                  // Nonstatic class variables
   
  public Complex ()  { re = 0;  im = 0; }                      // Default constructor
   
  public Complex(double x, double y)  { re = x;  im = y; }        // Full constructor
   
  public static Complex add(Complex a, Complex b) {              // Static add method
    Complex temp = new Complex();                              // Create Complex temp
    temp.re = a.re + b.re; 
    temp.im = a.im + b.im; 
    return temp; 
  }
      
  public static Complex mult(Complex a, Complex b) {            // Static mult method
    Complex temp = new Complex();                              // Create Complex temp
    temp.re = a.re * b.re - a.im * b.im; 
    temp.im = a.re * b.im + a.im * b.re; 
    return temp;
  }
      
  public static void main(String[] argv) {                             // Main method
    Complex a, b;                                        // Declare 2 Complex objects
    a = new Complex();                                          // Create the objects
    b = new Complex(4.7, 3.2); 
    Complex c = new Complex(3.1, 2.4);                        // Declare, create in 1
    System.out.println("a,b=("+a.re+","+a.im+"), ("+b.re+","+b.im+"),");
    System.out.println("c = ("+c.re+ ", " +c.im+ ")"); 
    a = add(b, c);                                              // Perform arithmetic
    System.out.println( "b + c = (" +a.re+ ", " +a.im+ "),  "); 
    a = mult(b, c); 
    System.out.println( "b*c = (" +a.re+ ", " +a.im+ ")"); 
  }
}

File: OlderCodes/Fortran77codes/complex.f
c  From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
c  by RH Landau, MJ Paez, and CC BORDEIANU 
c  Copyright Princeton University Press, Princeton, 2008.
c  Electronic Materials copyright: R Landau, Oregon State Univ, 2008;
c  MJ Paez, Univ Antioquia, 2008; and CC BORDEIANU, Univ Bucharest, 2008.
c  Support by National Science Foundation   
c								                                             
c  complex.f: Dealing with complex numbers on a comput        
c 
      Program complex 
			
        Implicit none 
        Complex*16 z, zsqrt, zlog
       	Real*8 i, pi, phi, x, y, zatan, zatan2
       	pi = 3.1415926535897932385E0
       	Write (*,10) 'phi', 'x', 'y', 'sqrt', 'log', 'atan', 'atan2'
       	Write (*, *) ' '
c                                                      loop for angle
 	      Do 100 i = 0, 2.6, 0.1
          phi = i * pi
c                                 calculate carthesian representation
          x = cos(phi)
          y = sin(phi)
          z = cmplx(x, y)
          zsqrt = sqrt(z)
          zlog = log(z)
          zatan = atan(y/x)
          zatan2 = atan2(y, x)
          Write (*, 20) i, '*pi', x, y, zsqrt, 'i', zlog, 
					                                 'i', zatan, zatan2
 100   	Continue
 10    	Format (a4, 2a9, a14, a18, a14, a10)
 20    	Format (f3.1, a3, 3f9.4, f8.4, a1, f9.4, f8.4,a1, 2f9.4)
	      Stop 'complex'
      End
